### PR TITLE
feature/read_FT_Category_implementation

### PR DESCRIPTION
**Pull request by MackowiakPiotr**

### CODE REVIEW

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/model/FinancialTransactionCategory.java [hunk 1]
```diff
@@ -49,6 +51,8 @@ public class FinancialTransactionCategory implements Serializable {
     @Column(name = "creation_date")
     @DateTimeFormat(pattern = "yyyy-mm-dd hh:mm:ss")
     private Instant creationDate;
+    @OneToMany(mappedBy = "financialTransactionCategory")
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/model/FinancialTransactionCategory.java:54]: Insert one blank line below, please. Mean there should be an empty line:
- between every field in the class,
- between every method in the class,
- import and class declaration
- between the last method declaration character `}` and the end of the class declaration character `}`

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import javax.validation.constraints.NotNull;
+import java.math.BigInteger;
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+@Validated
+public class FinancialTransactionCategoryServiceImpl implements FinancialTransactionCategoryService {
+
+    private final FinancialTransactionCategoryRepository financialTransactionCategoryRepository;
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java:25]: Insert one blank line below, please. Mean there should be an empty line:

between every field in the class,
between every method in the class,
import and class declaration
between the last method declaration character } and the end of the class declaration character }

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Override
+    public List<FinancialTransactionCategoryDTO> getFinancialTransactionCategories() {
+        return financialTransactionCategoryRepository.findAll().stream()
+                .map(financialTransactionCategoryModelMapper::mapFinancialTransactionCategoryEntityToFinancialTransactionCategoryDTO)
+                .toList();
+    }
+
+    @Override
+    public FinancialTransactionCategoryDetailedDTO findById(@Min(1) @NotNull Long id) {
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java:37]: Name of variables refactoring suggestion:
```java
@Override
public FinancialTransactionCategoryDetailedDTO findById(@Min(1) @NotNull Long id) {       
	FinancialTransactionCategory ftcEntity = financialTransactionCategoryRepository.findById(id)
			.orElseThrow(() -> new AppRuntimeException(ErrorCode.FTC001,
					String.format("Financial transaction category with id: %d not found", id)));
					
	FinancialTransactionCategoryDTO ftcDTO = financialTransactionCategoryModelMapper
			.mapFinancialTransactionCategoryEntityToFinancialTransactionCategoryDTO(ftcEntity);

	BigInteger counter = BigInteger.valueOf(financialTransactionRepository.countFinancialTransactionsByFinancialTransactionCategoryId(id));

  return new FinancialTransactionCategoryDetailedDTO(ftcDTO, counter);
}
```

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Changes required.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/mapper/FinancialTransactionCategoryModelMapper.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package pl.byczazagroda.trackexpensesappbackend.mapper;
+
+import org.mapstruct.Mapper;
+import pl.byczazagroda.trackexpensesappbackend.dto.FinancialTransactionCategoryDTO;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionCategory;
```
[the remark this answers is not in the document]
Comment (MackowiakPiotr) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/mapper/FinancialTransactionCategoryModelMapper.java:5]: done

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/model/FinancialTransactionCategory.java (hunk 1 above)
Comment (MackowiakPiotr) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/model/FinancialTransactionCategory.java:55]: done, thank you

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/repository/FinancialTransactionCategoryRepository.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package pl.byczazagroda.trackexpensesappbackend.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionCategory;
```
[the remark this answers is not in the document]
Comment (MackowiakPiotr) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/repository/FinancialTransactionCategoryRepository.java:5]: done

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java (hunk 2 above)
Comment (MackowiakPiotr) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java:25]: done, thank you

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java (hunk 3 above)
Comment (MackowiakPiotr) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java]: Thank you, I corrected as you suggested

**Review by odzio33 — COMMENTED** [review]
Please add blank line between field in FinacialTransaction

**Review by odzio33 — APPROVED** [review]
Looks good for me

File: src/main/resources/db/changelog/007-add-foreignKey-to-financial_transaction.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+databaseChangeLog:
```
Comment (Krisu93) [inline src/main/resources/db/changelog/007-add-foreignKey-to-financial_transaction.yaml:1]: This changeset should by in file with creating financial transactions, no no in separate file

File: src/main/resources/db/changelog/data/financial_transactions.csv [hunk 7]
```diff
@@ -1,9 +1,12 @@
...
-1,1,INCOME,15000,2022-02-19T09:47:52.595721658,""
-2,2,EXPENSE,5000,2022-02-19T09:47:52.595721658,null
-3,2,EXPENSE,5000,2022-02-19T09:47:52.595721658,description
-4,2,EXPENSE,5000,2022-02-19T09:47:52.595721658,description
-5,3,INCOME,5000,2022-02-19T09:47:52.595721658,description
-6,3,INCOME,4000,2022-02-19T09:47:52.595721658,description
-7,4,INCOME,15000,2022-02-19T09:47:52.595721658,description
-8,4,INCOME,2000,2022-02-19T09:47:52.595721658,description
+id,wallet_id,transaction_type,amount,transaction_date,description,financial_transaction_category_id
+1,1,INCOME,15000,2022-02-19T09:47:52.595721658,"",""
```
Comment (Krisu93) [inline src/main/resources/db/changelog/data/financial_transactions.csv:2]: Field financial_transaction_category_id is number, not string

File: src/main/resources/db/dbchangelog-dev.yaml [hunk 8]
```diff
@@ -10,4 +10,6 @@ databaseChangeLog:
   - include:
       file: db/changelog/005-create-table-financial_transaction_categories.yaml
   - include:
-      file: db/changelog/006-insert-into-financial_transaction_categories.yaml
\ No newline at end of file
+      file: db/changelog/006-insert-into-financial_transaction_categories.yaml
+  - include:
+      file: db/changelog/007-add-foreignKey-to-financial_transaction.yaml
```
Comment (Krisu93) [inline src/main/resources/db/dbchangelog-dev.yaml:15]: Remove this one "include"

**Review by Krisu93 — CHANGES_REQUESTED** [review]

File: src/main/resources/db/changelog/data/financial_transactions.csv (hunk 7 above)
Comment (MackowiakPiotr) [reply src/main/resources/db/changelog/data/financial_transactions.csv]: done

**Pull request closed without merging**